# Introduction

This project is about analyzing a dataset of 87,489 audiobooks.The audiobook company wants to know what kind of book is most popular among readers,which author wrote and which narrator narrated books that received most 5-star ratings and how prices vary among different audiobooks.

The dataset comes from Kaggle at https://www.kaggle.com/datasets/padiamilind/audible-books/data

In [1]:
import pandas as pd
import numpy as np
from datetime import datetime

In [2]:
df=pd.read_csv(r"C:\Users\syuan\Downloads\audible.csv")

In [3]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 87489 entries, 0 to 87488
Data columns (total 8 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   name         87489 non-null  object
 1   author       87489 non-null  object
 2   narrator     87489 non-null  object
 3   time         87489 non-null  object
 4   releasedate  87489 non-null  object
 5   language     87489 non-null  object
 6   stars        87489 non-null  object
 7   price        87489 non-null  object
dtypes: object(8)
memory usage: 5.3+ MB


In [4]:
#Check if there are duplicates
df.duplicated().sum()

0

In [5]:
#Check if there's null
df.isnull().sum()

name           0
author         0
narrator       0
time           0
releasedate    0
language       0
stars          0
price          0
dtype: int64

In [6]:
df.head(10)

name                     author  \
0                  Geronimo Stilton #11 & #12  Writtenby:GeronimoStilton   
1                            The Burning Maze      Writtenby:RickRiordan   
2                                The Deep End       Writtenby:JeffKinney   
3                        Daughter of the Deep      Writtenby:RickRiordan   
4  The Lightning Thief: Percy Jackson, Book 1      Writtenby:RickRiordan   
5           The Hunger Games: Special Edition   Writtenby:SuzanneCollins   
6                 Quest for the Diamond Sword     Writtenby:WinterMorgan   
7                           The Dark Prophecy      Writtenby:RickRiordan   
8                   Merlin Mission Collection  Writtenby:MaryPopeOsborne   
9                         The Tyrant’s Tomb      Writtenby:RickRiordan   

                     narrator                time releasedate language  \
0       Narratedby:BillLobely   2 hrs and 20 mins    4-8-2008  English   
1    Narratedby:RobbieDaymond   13 hrs and 8 mins    1-5-2018  English   
2       Narratedby:DanRussell    2 hrs and 3 mins   6-11-2020  English   
3   Narratedby:SoneelaNankani  11 hrs and 16 mins   5-10-2021  English   
4   Narratedby:JesseBernstein              10 hrs   1-13-2010  English   
5   Narratedby:TatianaMaslany  10 hrs and 35 mins    30-10-18  English   
6      Narratedby:LukeDaniels   2 hrs and 23 mins    25-11-14  English   
7    Narratedby:RobbieDaymond  12 hrs and 32 mins    2-5-2017  English   
8  Narratedby:MaryPopeOsborne  10 hrs and 56 mins    2-5-2017  English   
9    Narratedby:RobbieDaymond  13 hrs and 22 mins    24-09-19  English   

                           stars     price  
0     5 out of 5 stars34 ratings       468  
1   4.5 out of 5 stars41 ratings       820  
2   4.5 out of 5 stars38 ratings       410  
3   4.5 out of 5 stars12 ratings       615  
4  4.5 out of 5 stars181 ratings       820  
5     5 out of 5 stars72 ratings       656  
6     5 out of 5 stars11 ratings       233  
7     5 out of 5 stars50 ratings       820  
8      5 out of 5 stars5 ratings  1,256.00  
9     5 out of 5 stars58 ratings       820

In [7]:
#Remove unnecessary words in columns

def replace_value(value):
    if 'Writtenby:' in value:
        return value.replace('Writtenby:',' ')
    elif 'Narratedby:' in value:
        return value.replace('Narratedby:',' ')
    else:
        return value

In [8]:
df['author']=df['author'].apply(replace_value)
df['narrator']=df['narrator'].apply(replace_value)

In [9]:
df.head()

name            author  \
0                  Geronimo Stilton #11 & #12   GeronimoStilton   
1                            The Burning Maze       RickRiordan   
2                                The Deep End        JeffKinney   
3                        Daughter of the Deep       RickRiordan   
4  The Lightning Thief: Percy Jackson, Book 1       RickRiordan   

          narrator                time releasedate language  \
0       BillLobely   2 hrs and 20 mins    4-8-2008  English   
1    RobbieDaymond   13 hrs and 8 mins    1-5-2018  English   
2       DanRussell    2 hrs and 3 mins   6-11-2020  English   
3   SoneelaNankani  11 hrs and 16 mins   5-10-2021  English   
4   JesseBernstein              10 hrs   1-13-2010  English   

                           stars price  
0     5 out of 5 stars34 ratings   468  
1   4.5 out of 5 stars41 ratings   820  
2   4.5 out of 5 stars38 ratings   410  
3   4.5 out of 5 stars12 ratings   615  
4  4.5 out of 5 stars181 ratings   820

In [10]:
#Insert space before last name

def insert_space_before_upper(n):
    output=''
    
    for i in n:
        if i.isupper():
            output=output+" "+i
        else:
            output=output+i
    return output

df['author']=df['author'].apply(insert_space_before_upper)
df['narrator']=df['narrator'].apply(insert_space_before_upper)


In [11]:
df.head()

name              author  \
0                  Geronimo Stilton #11 & #12    Geronimo Stilton   
1                            The Burning Maze        Rick Riordan   
2                                The Deep End         Jeff Kinney   
3                        Daughter of the Deep        Rick Riordan   
4  The Lightning Thief: Percy Jackson, Book 1        Rick Riordan   

            narrator                time releasedate language  \
0        Bill Lobely   2 hrs and 20 mins    4-8-2008  English   
1     Robbie Daymond   13 hrs and 8 mins    1-5-2018  English   
2        Dan Russell    2 hrs and 3 mins   6-11-2020  English   
3    Soneela Nankani  11 hrs and 16 mins   5-10-2021  English   
4    Jesse Bernstein              10 hrs   1-13-2010  English   

                           stars price  
0     5 out of 5 stars34 ratings   468  
1   4.5 out of 5 stars41 ratings   820  
2   4.5 out of 5 stars38 ratings   410  
3   4.5 out of 5 stars12 ratings   615  
4  4.5 out of 5 stars181 ratings   820

In [12]:
#Format releasedate column
df['releasedate'] = pd.to_datetime(df['releasedate'], format='mixed').dt.strftime('%Y-%m-%d')

In [13]:
df.head()

name              author  \
0                  Geronimo Stilton #11 & #12    Geronimo Stilton   
1                            The Burning Maze        Rick Riordan   
2                                The Deep End         Jeff Kinney   
3                        Daughter of the Deep        Rick Riordan   
4  The Lightning Thief: Percy Jackson, Book 1        Rick Riordan   

            narrator                time releasedate language  \
0        Bill Lobely   2 hrs and 20 mins  2008-04-08  English   
1     Robbie Daymond   13 hrs and 8 mins  2018-01-05  English   
2        Dan Russell    2 hrs and 3 mins  2020-06-11  English   
3    Soneela Nankani  11 hrs and 16 mins  2021-05-10  English   
4    Jesse Bernstein              10 hrs  2010-01-13  English   

                           stars price  
0     5 out of 5 stars34 ratings   468  
1   4.5 out of 5 stars41 ratings   820  
2   4.5 out of 5 stars38 ratings   410  
3   4.5 out of 5 stars12 ratings   615  
4  4.5 out of 5 stars181 ratings   820

In [14]:
#Extract stars and numbers of ratings and form new columns
df['ratings']=df['stars'].str.extract(r'(\d+) ratings')
df['stars']=df['stars'].str.extract(r'(\d+)')

In [15]:
df.head(50)

name                 author  \
0                          Geronimo Stilton #11 & #12       Geronimo Stilton   
1                                    The Burning Maze           Rick Riordan   
2                                        The Deep End            Jeff Kinney   
3                                Daughter of the Deep           Rick Riordan   
4          The Lightning Thief: Percy Jackson, Book 1           Rick Riordan   
5                   The Hunger Games: Special Edition        Suzanne Collins   
6                         Quest for the Diamond Sword          Winter Morgan   
7                                   The Dark Prophecy           Rick Riordan   
8                           Merlin Mission Collection      Mary Pope Osborne   
9                                 The Tyrant’s Tomb           Rick Riordan   
10           The Titan's Curse: Percy Jackson, Book 3           Rick Riordan   
11            Magic Tree House Collection: Books 9-16      Mary Pope Osborne   
12             Magic Tree House Collection: Books 1-8      Mary Pope Osborne   
13              Magnus Chase and the Ship of the Dead           Rick Riordan   
14                                    Northern Lights         Philip Pullman   
15                       Geronimo Stilton #13 and #14       Geronimo Stilton   
16                        Magic Tree House Collection      Mary Pope Osborne   
17                                              Exile      Shannon Messenger   
18                          Merlin Mission Collection      Mary Pope Osborne   
19                                          Neverseen      Shannon Messenger   
20                                  The Tower of Nero           Rick Riordan   
21              Eldest: The Inheritance Cycle, Book 2    Christopher Paolini   
22                                       Artemis Fowl            Eoin Colfer   
23                           Geronimo Stilton 22 & 24       Geronimo Stilton   
24      Percy Jackson and the Battle of the Labyrinth           Rick Riordan   
25                                    Winnie-the-Pooh            A. A. Milne   
26           Magic Tree House Collection: Books 25-32      Mary Pope Osborne   
27                       Geronimo Stilton #20 and #21       Geronimo Stilton   
28                                    The Red Pyramid           Rick Riordan   
29                      Mystery of the Griefer's Mark          Winter Morgan   
30                               Geronimo Stilton #17       Geronimo Stilton   
31                           The House at Pooh Corner            A. A. Milne   
32                                Mission Unstoppable             Dan Gutman   
33                                Darkness of Dragons      Tui T. Sutherland   
34  14 Short Stories Bundle for Kids: Adventure, S...        Nayomi Phillips   
35                            The Adventures of Rusty            Ruskin Bond   
36                       Geronimo Stilton #15 and #16       Geronimo Stilton   
37                                      The Lost Heir      Tui T. Sutherland   
38                                          Everblaze      Shannon Messenger   
39                        Treasure Hunters in Trouble          Winter Morgan   
40                                           Lodestar      Shannon Messenger   
41                           Curse of the Arctic Star          Carolyn Keene   
42                          The Ghost of Grey Fox Inn          Carolyn Keene   
43      Rowley Jefferson's Awesome Friendly Adventure            Jeff Kinney   
44                                          Flashback      Shannon Messenger   
45                            Secret of the Red Arrow      Franklin W. Dixon   
46         The Lightning Thief: Percy Jackson, Book 1           Rick Riordan   
47                              Clash of the Creepers          Winter Morgan   
48                          The Skeletons Strike Back          Winter Morgan   
49                                       Dragonsl

In [16]:
df['ratings']=df['ratings'].fillna(0)
df['stars']=df['stars'].fillna(0)

In [17]:
#Calculate duration
hour=df['time'].str.extract(r'(\d+)').astype('int')
mins=df['time'].str.extract(r'(\d+) mins')
mins=mins.fillna(0).astype('int')

In [18]:
print(hour)

        0
0       2
1      13
2       2
3      11
4      10
...    ..
87484   7
87485  10
87486  19
87487   4
87488   6

[87489 rows x 1 columns]


In [19]:
print(mins)

        0
0      20
1       8
2       3
3      16
4       0
...    ..
87484  34
87485   7
87486   4
87487   8
87488   0

[87489 rows x 1 columns]


In [20]:
duration=(hour*60)+mins
df['duration_min']=duration

In [21]:
df.head(10)

name               author  \
0                  Geronimo Stilton #11 & #12     Geronimo Stilton   
1                            The Burning Maze         Rick Riordan   
2                                The Deep End          Jeff Kinney   
3                        Daughter of the Deep         Rick Riordan   
4  The Lightning Thief: Percy Jackson, Book 1         Rick Riordan   
5           The Hunger Games: Special Edition      Suzanne Collins   
6                 Quest for the Diamond Sword        Winter Morgan   
7                           The Dark Prophecy         Rick Riordan   
8                   Merlin Mission Collection    Mary Pope Osborne   
9                         The Tyrant’s Tomb         Rick Riordan   

              narrator                time releasedate language stars  \
0          Bill Lobely   2 hrs and 20 mins  2008-04-08  English     5   
1       Robbie Daymond   13 hrs and 8 mins  2018-01-05  English     4   
2          Dan Russell    2 hrs and 3 mins  2020-06-11  English     4   
3      Soneela Nankani  11 hrs and 16 mins  2021-05-10  English     4   
4      Jesse Bernstein              10 hrs  2010-01-13  English     4   
5      Tatiana Maslany  10 hrs and 35 mins  2018-10-30  English     5   
6         Luke Daniels   2 hrs and 23 mins  2014-11-25  English     5   
7       Robbie Daymond  12 hrs and 32 mins  2017-02-05  English     5   
8    Mary Pope Osborne  10 hrs and 56 mins  2017-02-05  English     5   
9       Robbie Daymond  13 hrs and 22 mins  2019-09-24  English     5   

      price ratings  duration_min  
0       468      34           140  
1       820      41           788  
2       410      38           123  
3       615      12           676  
4       820     181           600  
5       656      72           635  
6       233      11           143  
7       820      50           752  
8  1,256.00       5           656  
9       820      58           802

In [22]:
#Rename columns
df=df.rename(columns={'name':'Title','author':'Author','narrator':'Narrator','time':'Duration','releasedate':'Releasedate','language':'Language','stars':'Stars','ratings':'Ratings','duration_min':'Duration_min','price':'Price'})
df=df[['Title','Author','Narrator','Duration','Duration_min','Releasedate','Language','Stars','Ratings','Price']]

In [23]:
df.head()

Title              Author  \
0                  Geronimo Stilton #11 & #12    Geronimo Stilton   
1                            The Burning Maze        Rick Riordan   
2                                The Deep End         Jeff Kinney   
3                        Daughter of the Deep        Rick Riordan   
4  The Lightning Thief: Percy Jackson, Book 1        Rick Riordan   

            Narrator            Duration  Duration_min Releasedate Language  \
0        Bill Lobely   2 hrs and 20 mins           140  2008-04-08  English   
1     Robbie Daymond   13 hrs and 8 mins           788  2018-01-05  English   
2        Dan Russell    2 hrs and 3 mins           123  2020-06-11  English   
3    Soneela Nankani  11 hrs and 16 mins           676  2021-05-10  English   
4    Jesse Bernstein              10 hrs           600  2010-01-13  English   

  Stars Ratings Price  
0     5      34   468  
1     4      41   820  
2     4      38   410  
3     4      12   615  
4     4     181   820

In [24]:
#Change data type
df['Releasedate']=pd.to_datetime(df['Releasedate'])
df['Stars']=df['Stars'].astype('int')
df['Ratings']=df['Ratings'].astype('int')


In [25]:
df['Price']=df['Price'].str.replace(',','')
df['Price']=pd.to_numeric(df['Price'],errors='coerce')
df['Price'].fillna(0,inplace=True)
df['Price']=df['Price'].astype('int')

In [26]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 87489 entries, 0 to 87488
Data columns (total 10 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   Title         87489 non-null  object        
 1   Author        87489 non-null  object        
 2   Narrator      87489 non-null  object        
 3   Duration      87489 non-null  object        
 4   Duration_min  87489 non-null  int32         
 5   Releasedate   87489 non-null  datetime64[ns]
 6   Language      87489 non-null  object        
 7   Stars         87489 non-null  int32         
 8   Ratings       87489 non-null  int32         
 9   Price         87489 non-null  int32         
dtypes: datetime64[ns](1), int32(4), object(5)
memory usage: 5.3+ MB


In [27]:
df.describe()

Duration_min                    Releasedate         Stars  \
count  87489.000000                          87489  87489.000000   
mean     652.637737  2018-06-22 01:35:29.780886528      0.742013   
min       60.000000            1998-12-27 00:00:00      0.000000   
25%      289.000000            2016-08-30 00:00:00      0.000000   
50%      480.000000            2020-01-30 00:00:00      0.000000   
75%      717.000000            2021-08-04 00:00:00      0.000000   
max     8595.000000            2025-11-14 00:00:00      5.000000   
std      648.941548                            NaN      1.657892   

            Ratings         Price  
count  87489.000000  87489.000000  
mean       2.605265    559.009224  
min        0.000000      0.000000  
25%        0.000000    268.000000  
50%        0.000000    585.000000  
75%        0.000000    755.000000  
max      985.000000   7198.000000  
std       26.386287    336.096631

In [28]:
star_counts=df.groupby('Stars')['Title'].count()
star_counts

Stars
0    72417
1      183
2      229
3     1124
4     6775
5     6761
Name: Title, dtype: int64

In [29]:
fivestar_titles = df[df['Stars'] == 5]
sorted=fivestar_titles.sort_values(by='Ratings',ascending=False)
unique=sorted.drop_duplicates(['Title','Narrator'])
unique.head(10)

Title  \
44433                                  Project Hail Mary   
7559     Harry Potter and the Chamber of Secrets, Book 2   
48143         Sherlock Holmes: The Definitive Collection   
64428                                      Wings of Fire   
23014                             The Ride of a Lifetime   
79754   Harry Potter and the Prisoner of Azkaban, Book 3   
7557         Harry Potter and the Goblet of Fire, Book 4   
7552   Harry Potter and the Order of the Phoenix, Book 5   
19431                                               Will   
62272                                      Atomic Habits   

                                                Author  \
44433                                        Andy Weir   
7559                                     J. K. Rowling   
48143    Stephen Fry-introductions, Arthur Conan Doyle   
64428                   A P J Abdul Kalam, Arun Tiwari   
23014                                      Robert Iger   
79754                                    J. K. Rowling   
7557                                     J. K. Rowling   
7552                                     J. K. Rowling   
19431                          Will Smith, Mark Manson   
62272                                      James Clear   

                           Narrator            Duration  Duration_min  \
44433                    Ray Porter  16 hrs and 10 mins           970   
7559                    Stephen Fry   11 hrs and 5 mins           665   
48143                   Stephen Fry  71 hrs and 57 mins          4317   
64428                 Girish Karnad    1 hr and 15 mins            75   
23014    Robert Iger, Jim Frangione   8 hrs and 45 mins           525   
79754                   Stephen Fry  13 hrs and 10 mins           790   
7557                    Stephen Fry  22 hrs and 17 mins          1337   
7552                    Stephen Fry  30 hrs and 18 mins          1818   
19431                    Will Smith  16 hrs and 16 mins           976   
62272                   James Clear   5 hrs and 35 mins           335   

      Releasedate Language  Stars  Ratings  Price  
44433  2021-04-05  English      5      926   1003  
7559   2015-11-20  English      5      845    999  
48143  2017-02-27  English      5      838   3037  
64428  2020-01-04  English      5      827     75  
23014  2019-09-23  English      5      819    888  
79754  2015-11-20  English      5      721    999  
7557   2015-11-20  English      5      664   1599  
7552   2015-11-20  English      5      636   1599  
19431  2021-09-11  English      5      591   1093  
62272  2018-10-18  English      5      573    820

#### The top 10 5-star audiobooks that receive most ratings are all English books and they are released between 2015 and 2021.7out of 10 of them are fiction stories.Two are autobiographies and other one is for personal development.

In [30]:
#Find out top5 audiobooks that got 5 stars and most ratings
fivestar_titles = df[df['Stars'] == 5]
top5title=fivestar_titles.nlargest(6,'Ratings').drop_duplicates('Title')['Title']
top5title

44433                                  Project Hail Mary
7559     Harry Potter and the Chamber of Secrets, Book 2
48143         Sherlock Holmes: The Definitive Collection
64428                                      Wings of Fire
23014                             The Ride of a Lifetime
Name: Title, dtype: object

In [31]:
#Find out top5 authors whose books got 5 stars and most ratings
best_author = df[df['Stars'] == 5]
top5author=best_author.nlargest(6,'Ratings').drop_duplicates('Author')['Author']
top5author

44433                                          Andy Weir
7559                                       J. K. Rowling
48143      Stephen Fry-introductions, Arthur Conan Doyle
64428                     A P J Abdul Kalam, Arun Tiwari
23014                                        Robert Iger
Name: Author, dtype: object

In [32]:
#Find out top5 narrators that got 5 stars and most ratings
best_narrator=df[df['Stars']==5]
top5narrator=best_narrator.nlargest(11,'Ratings').drop_duplicates('Narrator')['Narrator']
top5narrator

44433                      Ray Porter
7559                      Stephen Fry
64428                   Girish Karnad
23014      Robert Iger, Jim Frangione
19431                      Will Smith
Name: Narrator, dtype: object

In [33]:
fivestar_titles.describe()

Duration_min                    Releasedate   Stars      Ratings  \
count   6761.000000                           6761  6761.0  6761.000000   
mean     626.153971  2016-11-29 17:02:45.596805120     5.0     6.847360   
min       60.000000            1998-12-27 00:00:00     5.0     0.000000   
25%      327.000000            2014-07-25 00:00:00     5.0     0.000000   
50%      497.000000            2018-06-11 00:00:00     5.0     0.000000   
75%      722.000000            2020-05-03 00:00:00     5.0     3.000000   
max     7591.000000            2022-12-03 00:00:00     5.0   926.000000   
std      550.790347                            NaN     0.0    41.105922   

             Price  
count  6761.000000  
mean    692.392693  
min       0.000000  
25%     500.000000  
50%     668.000000  
75%     873.000000  
max    3037.000000  
std     321.337190

#### The average price of 5-star audiobooks is higher than that of all audiobooks  by 133.

In [34]:
df[(df['Price']==0)&(df['Stars']==5)]

Title  \
12588                       Learn French: Word Power 101   
12641  Learn French with Innovative Language's Proven...   
12652                     Learn Russian - Word Power 101   
12709  Learn Italian with Innovative Language's Prove...   
12778                      Learn Chinese: Word Power 101   
12779  Learn Russian with Innovative Language's Prove...   
12857  Learn Thai with Innovative Language's Proven L...   
12888  Learn Arabic with Innovative Language's Proven...   
12891  Learn Korean with Innovative Language's Proven...   
13004             Learn Mexican Spanish - Word Power 101   
19998                                       Frank Muller   
20235                          My Experiments with Truth   
22175  Saty Ke Prayog [The Story of My Experiments wi...   
23071                           Bharat 2020 [India 2020]   
28478  Network Marketing Sawal Aapke Jawab Surya Sinh...   
35358                  Diabetes I and II (Hindi Edition)   
35997        Freud Manovishleshan [Freud Psychoanalysis]   
44739                     Geedad Ka Shikar [Jackal Hunt]   
47095   21 Shreshth Kahaniyan Prem Chand (Hindi Edition)   
47102  Meri Priya Kahaniyaan: Phanishwarnath Renu [My...   
47112                                   Right Behind You   
47126  Meri Priya Kahaniyaan: Pratibha Rai [My Favori...   
48085                                       Interviewing   
48145                          Sevasadan [Service House]   
48171                           Kamayani (Hindi Edition)   
48176                            Somnath (Hindi Edition)   
48188  Shatranj Ke Khiladi Aur Anya Kahaniyaan [Chess...   
48190  Boodhi Kaki Tatha Anya Natak [Budhi Kaki and O...   
48203  Shakespeare Ki Kahaniyaan [Shakespeare's Stories]   
48215                      Vayam Raksham (Hindi Edition)   
48247  Meri Priya Kahaniyaan: Premchand [My Favorite ...   
48255                         Viraj Bahu (Hindi Edition)   
48258                               Gaban [Embezzlement]   
48269                         Raghuvansh (Hindi Edition)   
48272                                Pride and Prejudice   
48276  Toba Tek Singh Aur Anya Kahaniyaan [Toba Tek S...   
48291    Kaili Kamini Aur Anita [Kylie Kamini and Anita]   
48333                             Badi Didi [Big Sister]   
48404                  Mandir Ki Nartaki [Temple Dancer]   
49828                               Dharamputra [Godson]   
50042                                 Wakalat [Advocacy]   
50340                                     Andhera [Dark]   
50575                              Lekha (Hindi Edition)   
55205  Khamoshi Ke Aanchal Mein [In the Midst of Sile...   
62453                                           Hinduism   
62458                                             Buddha   
62476                                               Gita   
62511       Aap Kya Nahin Kar Sakte [What Can't You Do?]   
62554                                 Safalta Ka Rahasya   
64526               Raat Bhaari Hai [The Night Is Heavy]   
66182  Encyclical Letter Laudato Si' of the Holy Fath...   
66889                         Shiv Puran (Hindi Edition)   
66895                                            Rigveda   
66896                                       Garuda Puran   
66898                        Kurma Puran (Hindi Edition)   
66901                        Kalki Puran (Hindi Edition)   
66902                                 Devi Bhagwat Puran   
66903           Kya Kahate Hain Puran [What Purana Said]   
71404  The Girl He Never Noticed, Part 1 (Hindi Edition)   
71408  The Girl He Never Noticed, Part 2 (Hindi Edition)   

                               Author  \
12588    Innovative Language Learning   
12641    Innovative Language Learning   
12652    Innovative Language Learning   
12709    Innovative Language Learning   
12778    Innovative Language Learning   
12779    Innovative Language Learning   
12857    Innovative Language Learning   
12888    Innovative Language Learning   
12891    Innovative L

#### The 5-star audiobooks that are free are mostly Hindi books and foreign language learning books for English speakers.

In [35]:
df[df['Price']==3037]

Title  \
48143  Sherlock Holmes: The Definitive Collection   
63949                            Capital Volume 3   

                                                Author         Narrator  \
48143    Stephen Fry-introductions, Arthur Conan Doyle      Stephen Fry   
63949                                        Karl Marx    Derek Le Page   

                 Duration  Duration_min Releasedate Language  Stars  Ratings  \
48143  71 hrs and 57 mins          4317  2017-02-27  English      5      838   
63949   50 hrs and 4 mins          3004  2019-02-25  English      5        0   

       Price  
48143   3037  
63949   3037

In [36]:
df[df['Duration_min']==7591]

Title           Author  \
38010  The Decline and Fall of the Roman Empire    Edward Gibbon   

                 Narrator             Duration  Duration_min Releasedate  \
38010    Charlton Griffin  126 hrs and 31 mins          7591  2015-04-21   

      Language  Stars  Ratings  Price  
38010  English      5        2   1338

#### The duration and prices vary a lot among 5-star audibooks.The most expensive 5-star audiobooks are 'Sherlock Holmes: The Definitive Collection' and 'Capital Volume 3' and their duration is 4317 mins and 3004 mins respectively.The 5-star audiobook with longest duration of 7591 mins is 'The Decline and Fall of the Roman Empire'.

In [37]:
#Check how strong Price is correlated with Duration
np.corrcoef(df['Duration_min'], df['Price'])

array([[ 1.        , -0.02686865],
       [-0.02686865,  1.        ]])

In [38]:
#Check how strong Price is correlated with stars received
np.corrcoef(df['Stars'], df['Price'])

array([[1.        , 0.17605878],
       [0.17605878, 1.        ]])

# Conclusion

1)The average price of 5-star audiobooks is higher than that of all audiobooks  by 133.

2)The top 10 5-star audiobooks that receive most ratings are all English books and they are released between 2015 and 2021.The top 10 5-star audiobooks that receive most ratings are all English books and they are released between 2015 and 2021.7out of 10 of them are fiction stories.Two are autobiographies and other one is for personal development.

3)The duration and prices vary a lot among 5-star audibooks.The most expensive 5-star audiobooks are 'Sherlock Holmes: The Definitive Collection' and 'Capital Volume 3' and their duration is 4317 mins and 3004 mins respectively.The 5-star audiobook with longest duration of 7591 mins is 'The Decline and Fall of the Roman Empire'.

4)The 5-star audiobooks that are free are mostly Hindi books and foreign language learning books for English speakers.

5)There is no strong correlation between price and duration or stars received.

In [39]:
df.to_csv(r"C:\Users\syuan\Downloads\audiobooks_cleaned.csv")